skfda/preprocessing/dim_reduction/feature_extraction/_fda_feature_union.py

```diff
@@ -0,0 +1,130 @@
+            of the tuple is name affected to the transformer while the
+            second element is a scikit-learn transformer instance.
+            The transformer instance can also be `"drop"` for it to be
+            ignored.
+        n_jobs:
+            Number of jobs to run in parallel.
+            ``None`` means 1 unless in a :obj:`joblib.parallel_backend`
+            context.
+            ``-1`` means using all processors.
+            The default value is None
+        transformer_weights:
+            Multiplicative weights for features per transformer.
+            Keys are transformer names, values the weights.
+            Raises ValueError if key not present in ``transformer_list``.
+        verbose:
+            If True, the time elapsed while fitting each transformer will be
+            printed as it is completed.
+        np_array_output:
+            indicates if the transformed data is requested to be a NumPy array
+            output. By default the value is False.
+
+    Examples:
+    Firstly we will import the Berkeley Growth Study data set
+    >>> from skfda.datasets import fetch_growth
+    >>> X, y= fetch_growth(return_X_y=True, as_frame=True)
+    >>> X = X.iloc[:, 0].values
+
+    Then we need to import the transformers we want to use
+    >>> from skfda.preprocessing.dim_reduction.feature_extraction import FPCA
+    >>> from skfda.representation import EvaluationTransformer
+
+    Finally we import the union and apply fit and transform
+    >>> from skfda.preprocessing.dim_reduction.feature_extraction.
+    ... _fda_feature_union import FdaFeatureUnion
+    >>> union = FdaFeatureUnion([
+    ...    ("Eval", EvaluationTransformer()),
+    ...    ("fpca", FPCA()), ], np_array_output=True)
+    >>> union.fit_transform(X)
+    """
+
+    def __init__(
+        self,
+        transformer_list,
+        *,
+        n_jobs=None,
+        transformer_weights=None,
+        verbose=False,
+        np_array_output=False,
+    ) -> None:
+        self.np_array_output = np_array_output
+        super().__init__(
+            transformer_list,
+            n_jobs=n_jobs,
+            transformer_weights=transformer_weights,
+            verbose=verbose,
+        )
+
+    def _hstack(self, Xs):
+
+        if (self.np_array_output):
+            for i in Xs:
+                if isinstance(i, FDataGrid or FDataBasis):
+                    raise TypeError(
+                        "There are transformed instances of FDataGrid or "
+                        "FDataBasis that can't be concatenated on a NumPy "
+                        "array.",
+                    )
+            return super()._hstack(Xs)
+
+        first_grid = True
+        first_basis = True
+        for j in Xs:
+            if isinstance(j, FDataGrid):
+                if first_grid:
+                    curves = j
+                    first_grid = False
+                else:
+                    curves = curves.concatenate(j)
```

Comment:
We do not want to concatenate the curves as new samples. They are different variables of the SAME observation. We want to return a Pandas DataFrame with one column per functional or multivariate variable.

Reply:
Okay, I understood